test/e2e/tests/test_queue.py

```diff
@@ -14,84 +14,114 @@
-        )
-        k8s.create_custom_resource(ref, resource_data)
-        cr = k8s.wait_resource_consumed_by_controller(ref)
+    resources = get_bootstrap_resources()
+    logging.debug(resources)
 
-        assert cr is not None
-        assert k8s.get_resource_exists(ref)
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["QUEUE_NAME"] = resource_name
 
-        time.sleep(CREATE_WAIT_AFTER_SECONDS)
+    # Load Queue CR
+    resource_data = load_sqs_resource(
+        "queue",
+        additional_replacements=replacements,
+    )
+    logging.debug(resource_data)
+
+    # Create k8s resource
+    ref = k8s.CustomResourceReference(
+        CRD_GROUP, CRD_VERSION, RESOURCE_PLURAL,
+        resource_name, namespace="default",
+    )
+    k8s.create_custom_resource(ref, resource_data)
+    cr = k8s.wait_resource_consumed_by_controller(ref)
 
-        cr = k8s.wait_resource_consumed_by_controller(ref)
+    sqsqueue.wait_until_exists(resource_name)
 
-        # Check Queue exists
-        exists = self.queue_exists(sqs_client, resource_name)
-        assert exists
+    yield cr, ref
 
-        # Delete k8s resource
-        _, deleted = k8s.delete_custom_resource(ref)
-        assert deleted is True
+    # Delete k8s resource
+    _, deleted = k8s.delete_custom_resource(
+        ref,
+        period_length=DELETE_WAIT_AFTER_SECONDS,
+    )
+    assert deleted
 
-        time.sleep(DELETE_WAIT_AFTER_SECONDS)
+    sqsqueue.wait_until_deleted(resource_name)
 
-        # Check Queue doesn't exist
-        exists = self.queue_exists(sqs_client, resource_name)
-        assert not exists
\ No newline at end of file
+
+@service_marker
+@pytest.mark.canary
+class TestQueue:
+    def test_crud(self, simple_queue):
+        res, ref = simple_queue
+
+        time.sleep(CREATE_WAIT_AFTER_SECONDS)
+
+        # Update the tags associated with the Queue and verify the update is
+        # reflected in the SQS API
+        cr = k8s.get_resource(ref)
+        assert cr is not None
+        assert 'spec' in cr
+        assert 'delaySeconds' in cr['spec']
+        assert cr['spec']['delaySeconds'] == "0"
+        assert 'status' in cr
+        assert 'queueURL' in cr['status']
+        queue_url = cr['status']['queueURL']
+
+        latest_attrs = sqsqueue.get_attributes(queue_url)
+        assert 'DelaySeconds' in latest_attrs
+        assert latest_attrs['DelaySeconds'] == "0"
+
+        # Test updating tags...
+        assert 'tags' in cr['spec']
+        assert len(cr['spec']['tags']) == 1
+        assert 'key1' in cr['spec']['tags']
+        assert cr['spec']['tags']['key1'] == 'val1'
+
+        expect_before_update_tags = {
+            "key1": "val1",
+        }
+        latest_tags = sqsqueue.get_tags(queue_url)
+        tags.assert_equal_without_ack_tags(
+            expect_before_update_tags, latest_tags,
+        )
+
+        new_tags = {
+            "key1": None, # Unfortunately, this is the only way to remove a key
```

Comment:
Wouldn't this cause `ev, found := existingTags[k]; !found || *ev != *v` in hooks.go to panic?

Reply:
Looks like it's not panicking ... Woundn't `None` translate to `nil` in go?

Reply:
No, this is a `kubectl patch|apply` peculiarity... this is the only way to tell kubectl you want to remove that key from the tags field.

Reply:
otherwise you need to mess with all the strategic merge patch foo.

Reply:
Patching system god lord bless